src/service/transaction-fees-indexer/persistance.ts

```diff
@@ -0,0 +1,46 @@
+import { MerkleTreeData } from './types';
+import { writeFile, readFile, mkdir } from 'fs/promises';
+import * as path from 'path';
+
+const dir = path.join(__dirname, './merkle-trees');
+
+const constructFilePath = ({
+  chainId,
+  epochNum,
+}: {
+  chainId: number;
+  epochNum: number;
+}) => path.join(dir, `merkletree-chain-${chainId}-epoch-${epochNum}.json`);
+
+export async function saveMerkleTree({
+  chainId,
+  epochNum,
+  merkleTree,
+}: {
+  chainId: number;
+  epochNum: number;
+  merkleTree: MerkleTreeData | null;
+}) {
+  const fileLocation = constructFilePath({ chainId, epochNum });
+
+  try {
+    await mkdir(dir, {
+      recursive: true,
+    });
+  } catch {}
+  await writeFile(fileLocation, JSON.stringify(merkleTree));
```

Comment:
might as well put this in the try block too.

Reply:
That's the very last step so no need to I'd say